### PR TITLE
BaSP-MR-42: Update schema and validation.

### PR DESCRIPTION
**Pull request by Flor-Farruggia**
Update to schema and validation in order to populated trainers and activity in the class files.

### CODE REVIEW

**Review by enriqueanastasini — COMMENTED** [review]
I think that for this to be seen in the GET you have to add the populate in the controller of CLASS

File: src/models/Class.js [hunk 1]
```diff
@@ -14,12 +14,13 @@ const classSchema = new Schema({
     required: true,
   },
   trainer: {
-    type: Schema.Types.ObjectId,
+    type: [Schema.Types.ObjectId],
```
Comment (EmaBaldor) [inline src/models/Class.js:17]: It does not need to be in brackets, it is not an array.

File: src/models/Class.js [hunk 2]
```diff
@@ -14,12 +14,13 @@ const classSchema = new Schema({
...
+    ref: 'trainer',
     required: true,
-
   },
   activity: {
-    type: Schema.Types.ObjectId,
+    type: [Schema.Types.ObjectId],
```
Comment (EmaBaldor) [inline src/models/Class.js:22]: It does not need to be in brackets. Ok for trainer which is an array, but activity is a single identifier.

**Review by EmaBaldor — APPROVED** [review]

File: src/models/Class.js [hunk 3]
```diff
@@ -14,12 +14,13 @@ const classSchema = new Schema({
     required: true,
   },
   trainer: {
-    type: Schema.Types.ObjectId,
+    type: [Schema.Types.ObjectId],
+    ref: 'trainer',
```
Comment (JuanMornik) [inline src/models/Class.js:18]: ref must be 'Trainer'

File: src/models/Class.js [hunk 4]
```diff
@@ -14,12 +14,13 @@ const classSchema = new Schema({
...
     required: true,
-
   },
   activity: {
     type: Schema.Types.ObjectId,
+    ref: 'activity',
```
Comment (JuanMornik) [inline src/models/Class.js:23]: ref must be 'Activity'

File: src/controllers/class.js [hunk 5]
```diff
@@ -20,6 +22,8 @@ const getClassById = (req, res) => {
   const { id } = req.params;
 
   classes.findById(id)
+    .populate('trainer')
```
Comment (JuanMornik) [inline src/controllers/class.js:25]: update populate reference with 'Trainer'

File: src/controllers/class.js [hunk 6]
```diff
@@ -2,6 +2,8 @@ const classes = require('../models/Class');
 
 const getClasses = (req, res) => {
   classes.find()
+    .populate('trainer')
```
Comment (JuanMornik) [inline src/controllers/class.js:5]: update populate reference with 'Trainer'

**Review by JuanMornik — CHANGES_REQUESTED** [review]

**Review by JuanMornik — APPROVED** [review]

**Review by enriqueanastasini — APPROVED** [review]

**Pull request merged**